kornia/enhance/adjust.py

```diff
@@ -297,6 +298,8 @@ def solarize(input: torch.Tensor, thresholds: Union[float, torch.Tensor] = 0.5,
             # TODO: I am not happy about this line, but no easy to do batch-wise operation
             additions = torch.stack([x.expand(*input.shape[1:]) for x in additions])
 
+        additions = additions.to(input.device).to(input.dtype)
+
```

Comment:
```suggestion
            additions = additions.to(input.device).to(input.dtype)
            additions = torch.stack([x.expand(*input.shape[1:]) for x in additions])
```
Same above.

No additional indent here or it might throw an error if ```additions``` is a float number.

Reply:
The problem is `additions` is cast to a tensor even in case it's a float, which is how I ran into this issue in the first place. So, either we remove that line (link below) or we can always treat it as a tensor.

[https://github.com/kornia/kornia/blob/ec0cf87659e4c8f168ddacbabf714a1d555b29e9/kornia/enhance/adjust.py#L281-L282](url)